robot/Cumulus/resources/CustomizableRollupsPageObject.py

```diff
@@ -43,12 +43,62 @@ def is_setting_present(self, object, name):
             print("crlp setting already exists")
         return isPresent
 
+    def is_filter_present(self, filtername):
+        """ Search for the presence of an active crlp filter setting record already. Return a boolean value accordingly
+		"""
+        isPresent = False
+        filter = npsp_lex_locators['crlps']['filter_group'].format(filtername)
+        self.builtin.log_to_console(filter)
+        list_ele = self.selenium.get_webelements(filter)
+        p_count=len(list_ele)
+        self.builtin.log_to_console(p_count)
+        if p_count == 0:
+            print("Filter Group Not found")
+        else:
+            isPresent = True
+            print("Filter Group Already Exists")
+        return isPresent
+
+    @capture_screenshot_on_error
+    def create_new_filter_setting(self, *args, **kwargs):
+        """Creates a new filter setting by taking in all the filtering criteria(s)
+		"""
+        view_filter_locator = npsp_lex_locators['crlps']['button'].format("View Filter Groups")
+        new_filter_locator = npsp_lex_locators['crlps']['button'].format("New Filter Group")
+        success_toast = npsp_lex_locators['crlps']['success_toast']
+        add_filter_btn = npsp_lex_locators['crlps']['button'].format("Add")
+        self.selenium.wait_until_page_contains_element(view_filter_locator)
+        view_filter_button = self.selenium.get_webelement(view_filter_locator)
+        self.selenium.click_element(view_filter_button)
+        self.selenium.wait_until_page_contains_element(new_filter_locator,60)
+        if self.is_filter_present(kwargs['Name']):
+            return
+        else:
+            self.selenium.click_element(new_filter_locator)
+            self.selenium.wait_until_page_contains_element(add_filter_btn,60)
+            self.npsp.populate_modal_form(**kwargs)
+            self.add_filter(*args)
+            self.selenium.click_button("Save")
+            self.selenium.wait_until_element_is_not_visible(success_toast)
+
+
+    @capture_screenshot_on_error
+    def add_filter(self, *args):
+        """Gets the filter criteria and adds into the filter modal
+		"""
+        modal_save_btn = npsp_lex_locators['crlps']['modal-button']
+        for arg in args:
+            add_filter_btn = npsp_lex_locators['crlps']['button'].format("Add")
+            self.salesforce._jsclick(add_filter_btn)
+            self.npsp.populate_modal_form(**arg)
+            self.salesforce._jsclick(modal_save_btn)
+
     @capture_screenshot_on_error
     def create_new_rollup_setting(self, **kwargs):
         """ Wait for the Iframe to be available and switch to the Frame.
-                    Confirm that a rollup setting of the same type does not exist
-                        Click on new and create a new rollup settings record
-                """
+            Confirm that a rollup setting of the same type does not exist
+            Click on new and create a new rollup settings record
+        """
```

Comment:
On line 108 instead of is_setting_present keyword can u use 'self._check_rollup_status' as this keyword includes check for active status as well

Reply:
Tried  self._check_rollup_status but the test fails. I will explain why in the meeting on monday

Reply:
sounds good, I will troubleshoot this after u merge ur PR